### PR TITLE
Various minor refactoring of menus

### PR DESCRIPTION
**Pull request by Robyt3**
## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/menus.h [hunk 1]
```diff
@@ -637,7 +637,6 @@ class CMenus : public CComponent
 		PAGE_KOG_LEGACY, // removed, redirects to PAGE_INTERNET
 		PAGE_DEMOS,
 		PAGE_SETTINGS,
-		PAGE_SYSTEM,
```
Comment (heinrich5991) [inline src/game/client/components/menus.h:640]: Does this have implications for backcompat of config variables? It looks like it, given the `PAGE_KOG_LEGACY` above.

Comment (Robyt3) [reply src/game/client/components/menus.h:640]: No, only the browser pages are saved to `ui_page`:

https://github.com/ddnet/ddnet/blob/349eb38ebfc3bbdf258774ab28d99a1c1fe257f5/src/game/client/components/menus.cpp#L2194-L2195

Comment (heinrich5991) [reply src/game/client/components/menus.h:640]: Perfect!

**Review by heinrich5991 — APPROVED** [review]
Thanks!

**Pull request merged**